report_py3o/models/py3o_report.py

```diff
@@ -198,6 +198,9 @@ def _get_parser_context(self, model_instance, data):
             report_xml._get_rendering_context(model_instance.ids, data)
         )
         context['objects'] = model_instance
+        # image false will crash base64 decoding
```

Comment:
@tbaden The image field does not exist on all models. It is in the template that you must add an `if' instruction before using this field (it is a demo template)

Reply:
how would this instruction look like in the odt template?
additional: wouldn't it be good to still check if image exists in objects and when false to set it to empty bytestring. otherwise you'll get an error if an user forgets this if instruction

Reply:
@tbaden A report can be defined for any model and the name of a field used to store an image is not always 'image'. 
It's important to manage this case into the template to also avoid to display an empty square if the image is missing into the document.
You can use a control structure into the template to avoid to output an image if not set https://py3otemplate.readthedocs.io/en/latest/templating.html#use-control-structures

Reply:
@lmignon  done

Reply:
@tbaden The demo template seems to no more work....

Reply:
how did you test this or whats the error?

Reply:
I functionally tests your changes thanks to ci/runbot... 
There is no more image into the generated report

Reply:
Ah I see. Do you have a working example for an if condition in a image frame option? I checked the reference you send, but the are only "normal" if conditions and not a single one for options. tried brackets for a single line if, but didn't work.

Reply:
@tbaden This template provides you some example on how to design a template .. https://bitbucket.org/faide/py3o.template/src/default/example/py3o_example_template.odt

Reply:
I think I finally figured it out.

Reply:
@lmignon is it really needed to dump the dict as there is no encoding involved?
https://github.com/OCA/reporting-engine/blob/47dc23626025b991388434e3bdb8add82854bcde/report_py3o_fusion_server/models/py3o_report.py#L72

json.dumps fails on bytestrings (images)

Reply:
IMO the problem is not there. This code works into production (V12) with complex reports involving images etc...

Reply:
I just cloned the v12 version of this module into my v12 odoo and get the same error when running tests:

> E       TypeError: b'iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAIAAACQd1PeAAAADElEQVQI12P4//8/AAX+Av7czFnnAAAAAElFTkSuQmCC' is not JSON serializable

If it's working on your odoo, maybe a missing package or something similar?

Reply:
strange, even tests on travis for V12 are OK. Should be investigated but unfortunately I've no time for it for the moment.

Reply:
@lmignon I'll investigate. Thanks a lot for the support so far.

Reply:
@lmignon the use of simplejson could fix this. do you see any drawbacks in this?

like: 

> import simplejson as json

Reply:
@tbaden Before introducing a new dependency I prefer to understand what's the root cause of the problem...

Reply:
I'll reset my dev env and install itagain. Maybe I messed it up somewhere for other developments

Reply:
I think we should try to stay with the standard python libraries. It doesn't seem normal to me that the result of serialization to json works with one library and not another when it shouldn't make any difference. I have the feeling that the problem encountered is elsewhere.

Reply:
@lmignon I found the "issue". In v11 the portal User has no image, in v12 the portal user has an image. If you delete the image in v12 you get the exact same error:

>  File "/usr/lib/python3.5/base64.py", line 46, in _bytes_from_decode_data
>     "string, not %r" % s.__class__.__name__) from None
> TypeError: argument should be a bytes-like object or ASCII string, not 'bool'

If I use my if structure template I get the same error:

> E TypeError: b'iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAIAAACQd1PeAAAADElEQVQI12P4//8/AAX+Av7czFnnAAAAAElFTkSuQmCC' is not JSON serializable

So it's a general issue in v11 and v12. As you said, this should be caught in the odt template with an if structure. Maybe my if structure is still faulty. Could you please check my odt [template](https://github.com/OCA/reporting-engine/raw/4bf56dfa1be1b861f4251f89cd8074614c0c7326/report_py3o/demo/res_user.odt)